vantage6/node/__init__.py

```diff
@@ -532,33 +524,38 @@ def connect_to_socket(self):
             new tasks.
         """
 
-        self.socketIO = SocketIO(
-            self.server_io.host,
-            port=self.server_io.port,
+        self.socketIO = SocketIO()
+        self.socketIO.connect(
+            url=f'{self.server_io.host}:{self.server_io.port}',
             headers=self.server_io.headers,
-            wait_for_connection=True
+            namespaces=['/tasks']
         )
 
         # define() returns the instantiated action_handler
-        self.socket_tasks = self.socketIO.define(NodeTaskNamespace, '/tasks')
-        self.socket_tasks.set_node_worker(self)
+        NodeTaskNamespace.node_worker_ref = self
+        self.socketIO.register_namespace(NodeTaskNamespace('/tasks'))
+        # self.socket_tasks = self.socketIO.define(, '/tasks')
+        # self.socket_tasks.set_node_worker(self)
 
         # Log the outcome
-        if self.socketIO.connected:
-            msg = 'connected to host={host} on port={port}'
-            msg = msg.format(
-                host=self.server_io.host,
-                port=self.server_io.port
-            )
-            self.log.info(msg)
+        while not self.socketIO.connected:
+            self.log.debug('waiting for socket connection...')
+            time.sleep(1)
+
+        msg = 'connected to host={host} on port={port}'
+        msg = msg.format(
+            host=self.server_io.host,
+            port=self.server_io.port
+        )
+        self.log.info(msg)
```

Comment:
The message could just be formatted as f-string:

`self.log.info(f'connected to host={self.server_io.host} on port={self.server_io.port}')`